src/components/selectionBoxLayer.ts

```diff
@@ -151,7 +191,8 @@ export module Components {
       }
       this._xScale = xScale;
       this._xScale.onUpdate(this._adjustBoundsCallback);
-      this._bindBoxDataValues();
+      this._syncPixelPositionAndValue();
+      this.render();
```

Comment:
What I mean is that the code smells. Does the sync method have to be called inside the renderImmediately()?